SPDX-FileCopyrightText: Copyright 2026 Daniel Hast

SPDX-License-Identifier: Apache-2.0

--- homebrew-install.sh
+++ homebrew-install.sh
@@ -182,9 +182,8 @@
   UNAME_MACHINE="$(uname -m)"
 
   # On Linux, this script installs to /home/linuxbrew/.linuxbrew only
-  HOMEBREW_PREFIX="/home/linuxbrew/.linuxbrew"
+  HOMEBREW_PREFIX="${PWD}/.linuxbrew"
   HOMEBREW_REPOSITORY="${HOMEBREW_PREFIX}/Homebrew"
-  HOMEBREW_CACHE="${HOME}/.cache/Homebrew"
 
   STAT_PRINTF=("/usr/bin/stat" "-c")
   PERMISSION_FORMAT="%a"
@@ -826,27 +825,6 @@
 fi
 execute_sudo "${CHOWN[@]}" "-R" "${USER}:${GROUP}" "${HOMEBREW_REPOSITORY}"
 
-if ! [[ -d "${HOMEBREW_CACHE}" ]]
-then
-  execute "${MKDIR[@]}" "${HOMEBREW_CACHE}"
-fi
-if exists_but_not_writable "${HOMEBREW_CACHE}"
-then
-  execute_sudo "${CHMOD[@]}" "g+rwx" "${HOMEBREW_CACHE}"
-fi
-if file_not_owned "${HOMEBREW_CACHE}"
-then
-  execute_sudo "${CHOWN[@]}" "-R" "${USER}" "${HOMEBREW_CACHE}"
-fi
-if file_not_grpowned "${HOMEBREW_CACHE}"
-then
-  execute_sudo "${CHGRP[@]}" "-R" "${GROUP}" "${HOMEBREW_CACHE}"
-fi
-if [[ -d "${HOMEBREW_CACHE}" ]]
-then
-  execute "${TOUCH[@]}" "${HOMEBREW_CACHE}/.cleaned"
-fi
-
 if should_install_command_line_tools && version_ge "${macos_version}" "10.13"
 then
   ohai "Searching online for the Command Line Tools"
@@ -956,7 +934,8 @@
   execute "${USABLE_GIT}" "-c" "init.defaultBranch=main" "init" "--quiet"
 
   # "git remote add" will fail if the remote is defined in the global config
-  execute "${USABLE_GIT}" "config" "remote.origin.url" "${HOMEBREW_BREW_GIT_REMOTE}"
+  # Patch: Temporarily set the remote to local repository
+  execute "${USABLE_GIT}" "config" "remote.origin.url" "${HOMEBREW_BREW_LOCAL_GIT_REMOTE}"
   execute "${USABLE_GIT}" "config" "remote.origin.fetch" "+refs/heads/*:refs/remotes/origin/*"
   execute "${USABLE_GIT}" "config" "--bool" "fetch.prune" "true"
 
@@ -977,6 +956,9 @@
 
   execute "${USABLE_GIT}" "remote" "set-head" "origin" "--auto" >/dev/null
 
+  # Patch: Set the remote to what it should be post-install
+  execute "${USABLE_GIT}" "config" "remote.origin.url" "${HOMEBREW_BREW_GIT_REMOTE}"
+
   LATEST_GIT_TAG="$("${USABLE_GIT}" -c "column.ui=never" tag --list --sort="-version:refname" | head -n1)"
   if [[ -z "${LATEST_GIT_TAG}" ]]
   then
@@ -1029,8 +1011,6 @@
     PATH_WARN=1
   fi
 
-  execute "${HOMEBREW_PREFIX}/bin/brew" "update" "--force" "--quiet"
-
   if [[ -n "${PATH_WARN-}" ]]
   then
     warn "${HOMEBREW_PREFIX}/bin is not in your PATH.
